tests/unit/core/test_nemo_callbacks.py

```diff
@@ -0,0 +1,212 @@
+# ! /usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright 2019 NVIDIA. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# =============================================================================
+
+import os
+import shutil
+from io import StringIO
+
+import pytest
+from tensorboard.backend.event_processing import event_file_inspector as efi
+from torch.utils.tensorboard import SummaryWriter
+
+from nemo.backends.pytorch.tutorials import MSELoss, RealFunctionDataLayer, TaylorNet
+from nemo.core.callbacks import *
+from nemo.utils import logging
+
+
+@pytest.mark.usefixtures("neural_factory")
+class TestNeMoCallbacks:
+    @pytest.fixture()
+    def clean_up(self):
+        yield
+        self.nf.reset_trainer()
+
+    @pytest.fixture()
+    def create_tensorboard_file(self):
+        os.makedirs("temp")
+        summary_writter = SummaryWriter("temp")
+        yield summary_writter
+        shutil.rmtree("temp")
+
+    @pytest.mark.unit
+    def test_SimpleLogger(self, clean_up):
+        data_source = RealFunctionDataLayer(n=100, batch_size=1)
+        trainable_module = TaylorNet(dim=4)
+        loss = MSELoss()
+
+        # Create the graph by connnecting the modules.
+        x, y = data_source()
+        y_pred = trainable_module(x=x)
+        loss_tensor = loss(predictions=y_pred, target=y)
+
+        # Mock up both std and stderr streams.
+        with logging.patch_stdout_handler(StringIO()) as std_out:
```

Comment:
ok, now I finally understand why you added those methods to logging... but should they really be part of logging, not testing?